### PR TITLE
Make max_sequence_length config check optional.

### PR DESCRIPTION
**Pull request by nverke**
This does not seem to get set for most cases since the config constructed by the model creation is never actually passed back out and therefore this value does not get set for llama models. 

Was getting while building my llama model
```
Traceback (most recent call last):
  File "/home/noah/anaconda3/envs/tvm-build-venv/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/noah/anaconda3/envs/tvm-build-venv/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/noah/workplace/mlc-llm/mlc_llm/build.py", line 13, in <module>
    main()
  File "/home/noah/workplace/mlc-llm/mlc_llm/build.py", line 10, in main
    core.build_model_from_args(parsed_args)
  File "/home/noah/workplace/mlc-llm/mlc_llm/core.py", line 578, in build_model_from_args
    mod = mod_transform_before_build(mod, param_manager, args, config)
  File "/home/noah/workplace/mlc-llm/mlc_llm/core.py", line 361, in mod_transform_before_build
    max_seq_len = config["max_sequence_length"]
KeyError: 'max_sequence_length'
```

### CODE REVIEW

**Comment by sudeepag** [comment]
This still results in an error when you don't explicitly pass in the `max-seq-len` arg: 

```shell
  File "/home/sudeepag/qa_latest/mlc-llm/mlc_llm/core.py", line 368, in mod_transform_before_build
    max_seq_len)
UnboundLocalError: local variable 'max_seq_len' referenced before assignment
```

**Comment by masahi** [comment]
oh right I've only checked vicuna 7B and 13B and thought all models have this config. But what is the expected behavior when both `max_sequence_length` and the cmd line `max-seq-len` are not provided?

File: mlc_llm/core.py [hunk 1]
```diff
@@ -358,7 +358,8 @@ def mod_transform_before_build(
     )  # pylint: disable=not-callable
 
     if "num_attention_heads" in config and "hidden_size" in config:
-        max_seq_len = config["max_sequence_length"]
+        if "max_sequence_length" in config:
+            max_seq_len = config["max_sequence_length"]
 
         if args.max_seq_len > 0:
             max_seq_len = args.max_seq_len
```
Comment (masahi) [inline mlc_llm/core.py:365]: How about 

```
if "max_sequence_length" in config:
    max_seq_len = config["max_sequence_length"]
elif args.max_seq_len > 0:
    max_seq_len = args.max_seq_len
else:
    max_seq_len = 2048
```
?

**Comment by nverke** [comment]
Updated to handle the case where the config is not set by the model with an error and also have the model pass back a model_config from the builder to be used for this.

File: mlc_llm/core.py (hunk 1 above)
Comment (nverke) [reply mlc_llm/core.py:364]: Is the intended behavior here that config["max_sequence_length"] takes precedent over args.max_seq_len I thought that was the other way around.

**Review by masahi — APPROVED** [review]
cc @Hzfengsy

**Review by MasterJH5574 — APPROVED** [review]

**Comment by sudeepag** [comment]
This change is causing an error when building Llama models:

```shell
Traceback (most recent call last):
  File "/home/sudeepag/.conda/envs/sudeep-mlc-llm/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/sudeepag/.conda/envs/sudeep-mlc-llm/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/sudeepag/qa_latest/mlc-llm/mlc_llm/build.py", line 13, in <module>
    main()
  File "/home/sudeepag/qa_latest/mlc-llm/mlc_llm/build.py", line 10, in main
    core.build_model_from_args(parsed_args)
  File "/home/sudeepag/qa_latest/mlc-llm/mlc_llm/core.py", line 583, in build_model_from_args
    mod = mod_transform_before_build(mod, param_manager, args, model_config)
  File "/home/sudeepag/qa_latest/mlc-llm/mlc_llm/core.py", line 362, in mod_transform_before_build
    if "num_attention_heads" in config and "hidden_size" in config:
TypeError: argument of type 'LlamaConfig' is not iterable
```

**Pull request merged**